kernel/kheap.c

```diff
@@ -9,6 +9,7 @@
 #include <wait_q.h>
 #include <init.h>
 
+#if defined(CONFIG_MULTITHREADING)
```

Comment:
dont like having this check all over the kernel, instead it would be nicer to have dedicated Kconfigs for the features being removed, so in this case I would not be building this file at all for example instead of doing selective conditionals, if that actually possible.
The benefit here being able to reduce footprint and remove unused code even with multithreading enabled...

Reply:
That is fine. We don't seem to use file-level conditional compilation in the kernel though, so I will instead follow the example of the other kernel files and rely on the linker.